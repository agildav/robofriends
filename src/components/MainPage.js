import React, { Component } from "react";
import CardList from "../components/CardList";
import SearchBox from "../components/SearchBox";

import "./MainPage.css";

class MainPage extends Component {
  filteredRobots = () => {
    return this.props.robots.filter(robot => {
      return robot.name
        .toLowerCase()
        .includes(this.props.searchfield.toLowerCase());
    });
  };

  render() {
    const { onSearchChange } = this.props;

    return (
      <div className="tc tracked">
        <h1 className="f1">Robofriends</h1>
        <SearchBox searchChange={onSearchChange} />
        <CardList robots={this.filteredRobots()} />
      </div>
    );
  }
}

export default MainPage;
