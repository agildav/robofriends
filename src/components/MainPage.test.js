import { shallow } from "enzyme";
import React from "react";
import MainPage from "./MainPage";

describe("MainPage test component", () => {
  let wrapper;

  beforeEach(() => {
    const mockProps = {
      robots: [],
      searchfield: "",
      onSearchChange: jest.fn()
    };

    wrapper = shallow(<MainPage {...mockProps} />);
  });

  test("MainPage container is rendered", () => {
    expect.assertions(1);
    return expect(wrapper).toMatchSnapshot();
  });

  test("Filters robots correctly", () => {
    expect.assertions(2);
    let mockProps2 = {
      robots: [
        {
          id: 1,
          name: "John",
          nickname: "Johnsi"
        }
      ],
      //  There is no robot with name 'a'
      searchfield: "a",
      onSearchChange: jest.fn()
    };
    //  No results
    let wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(wrapper2.instance().filteredRobots()).toEqual([]);

    mockProps2 = {
      robots: [
        {
          id: 1,
          name: "John",
          nickname: "Johnsi"
        }
      ],
      searchfield: "j",
      onSearchChange: jest.fn()
    };
    //  Got 1 result
    wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(wrapper2.instance().filteredRobots()).toEqual([
      {
        id: 1,
        name: "John",
        nickname: "Johnsi"
      }
    ]);
  });
});
