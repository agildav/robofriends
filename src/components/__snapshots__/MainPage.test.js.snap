// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage test component MainPage container is rendered 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    onSearchChange={[Function]}
    robots={Array []}
    searchfield=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="f1"
>
          Robofriends
</h1>,
        <SearchBox
          searchChange={[Function]}
/>,
        <CardList
          robots={Array []}
/>,
      ],
      "className": "tc tracked",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Robofriends",
          "className": "f1",
        },
        "ref": null,
        "rendered": "Robofriends",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "searchChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "robots": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="f1"
>
            Robofriends
</h1>,
          <SearchBox
            searchChange={[Function]}
/>,
          <CardList
            robots={Array []}
/>,
        ],
        "className": "tc tracked",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Robofriends",
            "className": "f1",
          },
          "ref": null,
          "rendered": "Robofriends",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "robots": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
